packages/datadog-instrumentations/src/graphql.js

```diff
@@ -235,14 +238,25 @@ function callInAsyncScope (fn, thisArg, args, abortController, cb) {
   try {
     const result = fn.apply(thisArg, args)
     if (result && typeof result.then === 'function') {
-      // bind callback to this scope
-      result.then(
-        res => cb(null, res),
-        err => cb(err)
+      // Mongoose queries are thenable, but not true promises. We can
+      // identify a mongoose query by the presence of an `exec` function and
+      // its constructor name, then wrap it.
+      if (typeof result.exec === 'function' && result.constructor.name === 'Query') {
```

Comment:
Is this enough to know that the object has come from mongoose? I doubt it is. Could we somehow tag it (e.g. by adding it to a WeakSet from within the mongoose integration) so that we know for _sure_ that's what's happening?